test/verify.test.js

```diff
@@ -307,77 +309,129 @@ test.serial('Verify "assets" is an Array of Object with a glob Arrays in path pr
   t.true(gitlab.isDone());
 });
 
-test('Throw SemanticReleaseError if "assets" option is not a String or an Array of Objects', async t => {
+test.serial('Throw SemanticReleaseError if "assets" option is not a String or an Array of Objects', async t => {
+  const owner = 'test_user';
+  const repo = 'test_repo';
   const env = {GITLAB_TOKEN: 'gitlab_token'};
   const assets = 42;
+  const gitlab = authenticate(env)
+    .get(`/projects/${owner}%2F${repo}`)
+    .reply(200, {permissions: {project_access: {access_level: 40}}});
 
-  const [error] = await t.throwsAsync(
+  const aggregateError = await t.throwsAsync(
     verify(
       {assets},
-      {env, options: {repositoryUrl: 'https://gitlab.com/semantic-release/gitlab.git'}, logger: t.context.logger}
-    )
+      {env, options: {repositoryUrl: `https://gitlab.com/${owner}/${repo}.git`}, logger: t.context.logger}
+    ),
+    AggregateError,
+    'Error thrown but of unexpected type.'
   );
+  const errors = [...aggregateError];
 
-  t.is(error.name, 'SemanticReleaseError');
-  t.is(error.code, 'EINVALIDASSETS');
+  const expectedNumOfErrors = 1;
+  t.is(errors.length, expectedNumOfErrors, `Expected ${expectedNumOfErrors} error but got ${errors.length}.`);
+  t.is(errors[0].name, 'SemanticReleaseError', 'Unexpected error name.');
```

Comment:
I feel the message `Unexpected error name.` is stating the obvious at beat and misleading at worst.
It's not that the error name is wrong, it's that a different error has been thrown.
It felt to me that the verification is so simple and the `ava` output does a really good job at indicating the assertion that failed and provide the correct/wrong values that adding a message here wasn't necessary.

Reply:
The reason I added failure message is explained about. Waiting for you feedback on the principle. 

As for the message itself, yes, I agree it is not the best one... What about something like `Unexpected error thrown` or `Error other than SemanticReleaseError thrown`?

![image](https://user-images.githubusercontent.com/1336689/67518330-e21afe00-f6a4-11e9-9df6-76d4d56e1f3f.png)